integration-tests/opentelemetry-reactive/src/main/java/io/quarkus/it/opentelemetry/reactive/EndUserResource.java

```diff
@@ -0,0 +1,109 @@
+package io.quarkus.it.opentelemetry.reactive;
+
+import jakarta.annotation.security.PermitAll;
+import jakarta.annotation.security.RolesAllowed;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.semconv.SemanticAttributes;
+
+@Path("/otel/enduser")
+public class EndUserResource {
+
+    @Inject
+    Tracer tracer;
+
+    @Path("/no-authorization")
+    @GET
+    public String noAuthorization() {
+        return "/no-authorization";
+    }
+
+    @RolesAllowed("WRITER")
+    @Path("/roles-allowed-only")
+    @GET
+    public String rolesAllowedOnly() {
+        return "/roles-allowed-only";
+    }
+
+    @PermitAll
+    @Path("/permit-all-only")
+    @GET
+    public String permitAllOnly() {
+        return "/permit-all-only";
+    }
+
+    @Path("/no-authorization-augmentor")
+    @GET
+    public String noAuthorizationAugmentor() {
+        return "/no-authorization-augmentor";
+    }
+
+    @RolesAllowed("PUBLISHER")
+    @Path("/roles-allowed-only-augmentor")
```

Comment:
I see that the test does augmentation if role is PUBLISHER... It's a bit confusing. 
Could you rename it to AUGMENTATOR... The endpoint would be `/roles-allowed-only--augmentator`.
Another idea would be to encode as a parameter the need to augment. This would reduce the nr of endpoints and make the code easier to read.

Reply:
sounds good. Added a new comment.